### PR TITLE
CHE-1231. Show/hide error messages after updating pom.xml

### PR DESCRIPTION
**Pull request by RomanNikitenko**
1. Display/hide error messages in the pom-file and opened java files after updating pom.xml
2. Fix showing duplicate error message in the pom.xml when a dependency is not found
3. Fix showing old error message in the pom.xml (problems with caching )

@vparfonov @evidolob please review

Signed-off-by: Roman Nikitenko rnikitenko@codenvy.com

### CODE REVIEW

File: plugins/plugin-maven/che-plugin-maven-server/src/main/java/org/eclipse/che/plugin/maven/server/rest/MavenServerService.java [hunk 1]
```diff
@@ -166,49 +166,39 @@ public Response reimportDependencies(@ApiParam(value = "The paths to projects wh
...
+            if (mavenProject == null) {
+                return result;
             }
+
+            List<MavenProjectProblem> problems = mavenProject.getProblems();
+            int start = pomContent.indexOf("<project ") + 1;
+            int end = start + "<project ".length();
+
+            List<Problem> problemList = problems.stream().map(mavenProjectProblem -> {
+                Problem problem = DtoFactory.newDto(Problem.class);
```
Comment (benoitf) [inline plugins/plugin-maven/che-plugin-maven-server/src/main/java/org/eclipse/che/plugin/maven/server/rest/MavenServerService.java]: shouldn't use .with**\* to chain dto ?

return DtoFactory.newDto(Problem.class).with...with...with

File: plugins/plugin-maven/che-plugin-maven-server/src/main/java/org/eclipse/che/plugin/maven/server/rest/MavenServerService.java [hunk 2]
```diff
@@ -166,49 +166,39 @@ public Response reimportDependencies(@ApiParam(value = "The paths to projects wh
     }
 
     @GET
-    @Path("pom/reconsile")
+    @Path("pom/reconcile")
     @Produces("application/json")
-    public List<Problem> reconsilePom(@QueryParam("pompath") String pomPath) {
+    public List<Problem> reconcilePom(@QueryParam("pompath") String pomPath) {
```
Comment (benoitf) [inline plugins/plugin-maven/che-plugin-maven-server/src/main/java/org/eclipse/che/plugin/maven/server/rest/MavenServerService.java]: I don't see swagger/javadoc about this REST service

File: plugins/plugin-maven/che-plugin-maven-server/src/main/java/org/eclipse/che/plugin/maven/server/rest/MavenServerService.java [hunk 3]
```diff
@@ -166,49 +166,39 @@ public Response reimportDependencies(@ApiParam(value = "The paths to projects wh
...
-                                        problem.setSourceStart(start);
-                                        problem.setSourceEnd(end);
-                                        problem.setMessage("Dependency " + artifact.getDisplayString() + " not found.");
-                                        return problem;
-                                    }).collect(Collectors.toList());
-
-                result.addAll(missedArtifacts);
-                result.addAll(problemList);
+            if (mavenProject == null) {
+                return result;
```
Comment (benoitf) [inline plugins/plugin-maven/che-plugin-maven-server/src/main/java/org/eclipse/che/plugin/maven/server/rest/MavenServerService.java:189]: this block is not tested

File: plugins/plugin-maven/che-plugin-maven-server/src/main/java/org/eclipse/che/plugin/maven/server/rest/MavenServerService.java [hunk 4]
```diff
@@ -166,49 +166,39 @@ public Response reimportDependencies(@ApiParam(value = "The paths to projects wh
...
             }
+
+            List<MavenProjectProblem> problems = mavenProject.getProblems();
```
Comment (benoitf) [inline plugins/plugin-maven/che-plugin-maven-server/src/main/java/org/eclipse/che/plugin/maven/server/rest/MavenServerService.java:192]: this block is not tested

File: plugins/plugin-maven/che-plugin-maven-server/src/main/java/org/eclipse/che/plugin/maven/server/rest/MavenServerService.java (hunk 1 above)
Comment (RomanNikitenko) [inline plugins/plugin-maven/che-plugin-maven-server/src/main/java/org/eclipse/che/plugin/maven/server/rest/MavenServerService.java]: @benoitf done!

File: plugins/plugin-maven/che-plugin-maven-server/src/main/java/org/eclipse/che/plugin/maven/server/rest/MavenServerService.java (hunk 2 above)
Comment (benoitf) [inline plugins/plugin-maven/che-plugin-maven-server/src/main/java/org/eclipse/che/plugin/maven/server/rest/MavenServerService.java]: thx

File: plugins/plugin-maven/che-plugin-maven-server/src/main/java/org/eclipse/che/plugin/maven/server/rest/MavenServerService.java (hunk 4 above)
Comment (benoitf) [inline plugins/plugin-maven/che-plugin-maven-server/src/main/java/org/eclipse/che/plugin/maven/server/rest/MavenServerService.java:192]: thx

**Comment by vparfonov** [comment]
ok

**Pull request merged**